tmk_core/common/action_layer.c

```diff
@@ -31,6 +32,12 @@ __attribute__((weak)) layer_state_t default_layer_state_set_kb(layer_state_t sta
  * Static function to set the default layer state, prints debug info and clears keys
  */
 static void default_layer_state_set(layer_state_t state) {
+    static bool has_run = false;
+    if (has_run) {
+        eeconfig_update_default_layer(state);
```

Comment:
Should this not be placed after the _kb and _user functions have processed it?

Reply:
Good point.  wasn't sure if that was the best behavior. But it probably is.